File: apps/moex_helper/frontend/scripts/helpers/i18n.js
import isString from 'lodash/isString'
import { Map } from 'immutable'

let translations = Map()

export const set = (trs) => {
  translations = trs
}

export const t = (path) => {
  if (isString(path)) path = path.split('.')
  return translations.getIn(path)
}

File: apps/moex_helper/lib/moex_helper/i18n.ex
defmodule MoexHelper.I18n do
  @translations %{
    en: %{
      security: %{
        data: %{
          secname: "Name",
          prevprice: "Last deal price",
          couponvalue: "Coupon value",
          nextcoupon: "Next coupon",
          matdate: "Maturity",
          lotsize: "Lot size"
        }
      }
    }
  }

  def t(path) when is_binary(path) do
    path |> String.split(".") |> Enum.map(&String.to_atom/1) |> t
  end

  def t(path) do
    get_in(@translations, path)
  end

  def translations do
    @translations
  end
end
